python-avd/pyavd/_eos_designs/shared_utils/mlag.py

```diff
@@ -214,3 +214,11 @@ def mlag_peer_ibgp_ip(self: SharedUtils) -> str:
     @cached_property
     def mlag_ibgp_peering_vrfs_base_vlan(self: SharedUtils) -> int:
         return int(get(self.hostvars, "mlag_ibgp_peering_vrfs.base_vlan", default=3000))
+
+    @cached_property
+    def mlag_peer_vrfs(self: SharedUtils) -> list[str]:
+        # Fact phase
+        if hasattr(self.mlag_peer_facts, "shared_utils"):
+            return self.mlag_peer_facts.shared_utils.local_vrfs
+        # structured_config phase
+        return self.get_mlag_peer_fact("local_vrfs")
```

Comment:
This would resolve to the same thing if called in both cases right? I mean get_mlag_peer_fact will get an instance of EosDesignsFacts or a dict. Both accept a `.get()` so I think you can skip the special handling here.

Reply:
indeed